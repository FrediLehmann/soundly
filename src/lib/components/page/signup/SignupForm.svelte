<script lang="ts" context="module">
  import { Button, Input } from '$lib/components/atoms';

  const Signupform = 'Signup';
</script>

<script lang="ts">
  export let signup: (email: string, password: string) => void;
  export let submitting: boolean;

  let emailError = '';
  let pwdError = '';

  const submit = (e: any) => {
    signup(e.target.email.value, e.target.password.value);
  };

  const click = async () => {
    const email = document.forms[Signupform].email;
    const pwd = document.forms[Signupform].password;

    if (email.validity.valid && pwd.validity.valid) {
      document.getElementById(`${Signupform}_submit`).click();
      return;
    }

    emailError = email.validationMessage;
    pwdError = pwd.validationMessage;
  };
</script>

<form on:submit|preventDefault={submit} name={Signupform}>
  <Input
    required
    type="email"
    name="email"
    label="Email"
    title="Please enter your email."
    error={emailError}
    disabled={submitting} />
  <Input
    required
    type="password"
    name="password"
    label="Password"
    title="Please enter your password."
    error={pwdError}
    disabled={submitting} />
  <Button disabled={submitting} on:click={click} style="primary" class="mbs-2">
    Sign up
  </Button>
  <input id={`${Signupform}_submit`} type="submit" hidden />
</form>

<style>
  form {
    display: flex;
    flex-direction: column;
    gap: 1rem;
  }
</style>
