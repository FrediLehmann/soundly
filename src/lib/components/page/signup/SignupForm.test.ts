/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import SignupForm from './SignupForm.svelte';

describe('<SignupForm /> component', () => {
  test('renders correctly', () => {
    const { component } = render(SignupForm, {
      props: { signup: (_: unknown) => _, submitting: false }
    });
    expect(component).toMatchSnapshot();
  });
});
