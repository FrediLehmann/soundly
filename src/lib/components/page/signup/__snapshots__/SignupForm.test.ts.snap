// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupForm /> component renders correctly 1`] = `
SignupForm {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      false,
      "",
      "",
      "",
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "signup": 6,
      "submitting": 0,
    },
    "root": <div>
      <form
        class="flex flex-col gap-3 svelte-1i4vv2l"
      >
        <div
          class="container svelte-1ulq3bj"
          data-required="true"
          data-sr-only="false"
        >
          <label
            class="svelte-1ulq3bj"
            for="email"
          >
            Email
          </label>
           
          <input
            class="svelte-1ulq3bj"
            id="email"
            required=""
            title="Please enter your email."
            type="email"
          />
           
        </div>
         
        <div
          class="container svelte-1ulq3bj"
          data-required="true"
          data-sr-only="false"
        >
          <label
            class="svelte-1ulq3bj"
            for="username"
          >
            Username
          </label>
           
          <input
            class="svelte-1ulq3bj"
            id="username"
            required=""
            title="Please enter a username."
            type="text"
          />
           
        </div>
         
        <div
          class="container svelte-1ulq3bj"
          data-required="true"
          data-sr-only="false"
        >
          <label
            class="svelte-1ulq3bj"
            for="password"
          >
            Password
          </label>
           
          <input
            class="svelte-1ulq3bj"
            id="password"
            required=""
            title="Please enter your password."
            type="password"
          />
           
        </div>
         
        <button
          class="mbs-2 svelte-1n3dce0"
          data-type="primary"
          type="button"
        >
          Sign up
        </button>
         
        <input
          hidden=""
          id="Signup_submit"
          type="submit"
        />
      </form>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
